File: frontend/lib/models/book_model.dart
class BookModel {
  String title;
  String author;
  String coverImagePath;
  String genre;
  String description;
  String pages;
  bool isFavorite;

  BookModel({
    required this.title,
    required this.author,
    required this.coverImagePath,
    required this.genre,
    required this.description,
    required this.pages,
    required this.isFavorite,
  });

  factory BookModel.fromJson(Map<String, dynamic> json) {
    return BookModel(
      title: json['title'] ?? '',
      author: json['author'] ?? '',
      coverImagePath: json['coverImagePath'] ?? '',
      genre: json['genre'] ?? '',
      description: json['description'] ?? '',
      pages: json['pages'] ?? '',
      isFavorite: json['isFavorite'] ?? false,
    );
  }
}

File: backend/models/bookmodel.js
const mongoose = require('mongoose');

const bookSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  author: {
    type: String,
    required: true,
  },
  coverImagePath: {   
    type: String,
    required: true,
  },
  genre: {
    type: String,
  },
  description: {
    type: String,
  },
  pages: {
    type: String,
  },
  isFavorite: {
    type: Boolean,
    default: false,
  },
  isPopular: {
    type: Boolean,
    default: false,
  },
});

const Book = mongoose.model('Book', bookSchema);

module.exports = Book;
